### ISSUE TITLE
updateParetoArchive does not consider bi-/multiobjective optimization directions

### ISSUE THREAD

**Issue by thllwg** [jakobbossek/ecr2#126]
ecr's `updateParetoArchive` function cannot handle different optimization goals in a multi-criteria case. The following example illustrates this:
```{r}
# stupid fitness function   
fitness.fun <- function(x){
    s = sum(x)
    return(c(s,s))
  }
# we want to maximise the first fitness value and minimize the second
ctrl = initECRControl(fitness.fun, n.objectives = 2L, minimize = c(FALSE, TRUE))
archive = initParetoArchive(ctrl)
population = list(c(1,1,0,1), c(0,0,0,1), c(1,1,1,1))
fitness = evaluateFitness(ctrl, population)
updateParetoArchive(archive, inds, fitness)
```
Fitness now contains three non-dominated points, but the pareto archive only contains individuals with minimized fitness
```{r}
> fitness
     [,1] [,2] [,3]
[1,]    3    1    4
[2,]    3    1    4
attr(,"class")
[1] "ecr_fitness_matrix" "matrix"            
attr(,"minimize")
[1] FALSE  TRUE
> # the pareto archive only contains individuals with minimized fitness
> getFront(archive)
     [,1]
[1,]    1
[2,]    1
```

**Comment by neverfox** [comment]
I think I'm noticing something similar. When I use `ecr` on a multi-objective with minimize `c(TRUE, FALSE)` I get back one non-dominating result, i.e. it appears to not take the one minimized dimension into account when determining domination and is doing both as a maximum. When I negate the first dimension and do both as maximize, I get the outcomes I would expect.

**Comment by jakobbossek** [comment]
Fixed via f4badde
